py3dtilers/Common/tileset_creation.py

```diff
@@ -67,20 +67,18 @@ def __transform_node(node: 'GeometryNode', user_args, tree_centroid=np.array([0,
             for feature_list in node.get_features():
                 feature_list.scale_features(user_args.scale, tree_centroid)
 
-        centroid = node.feature_list.get_centroid()
-        offset = np.array([0, 0, 0]) if user_args.offset[0] == 'centroid' else centroid + np.array(user_args.offset)
+        offset = np.array([0, 0, 0]) if user_args.offset[0] == 'centroid' else np.array(user_args.offset)
 
         if not user_args.crs_in == user_args.crs_out:
             transformer = Transformer.from_crs(user_args.crs_in, user_args.crs_out)
-            tree_centroid = np.array(transformer.transform(tree_centroid[0], tree_centroid[1], tree_centroid[2]))
-            offset = np.array(transformer.transform(offset[0], offset[1], offset[2]))
+            tree_centroid = np.array(transformer.transform((tree_centroid + offset)[0], (tree_centroid + offset)[1], (tree_centroid + offset)[2]))
             for feature_list in node.get_features():
-                feature_list.change_crs(transformer)
+                feature_list.change_crs(transformer, offset)
```

Comment:
Indeed, it seems a better and safer way to use offset at the same time as reprojection

A way to implement this improvement without modifying the `transform` behavior described below (in next comment) would be to do something like:

```python
        offset = np.array([0, 0, 0]) if user_args.offset[0] == 'centroid' else np.array(user_args.offset)
        transform_offset = node.feature_list.get_centroid() + offset

        if not user_args.crs_in == user_args.crs_out:
            transformer = Transformer.from_crs(user_args.crs_in, user_args.crs_out)
            tree_centroid = np.array(transformer.transform((tree_centroid + offset)[0], (tree_centroid + offset)[1], (tree_centroid + offset)[2]))
            for feature_list in node.get_features():
                feature_list.change_crs(transformer, offset)
            transform_offset = node.feature_list.get_centroid()

        distance = node.feature_list.get_centroid() - tree_centroid

        for feature_list in node.get_features():
            feature_list.translate_features(-feature_list.get_centroid())

        if user_args.obj is not None:
            for leaf in node.get_leaves():
                # Since the tiles are centered on [0, 0, 0], we use an offset to place the geometries in the OBJ model
                obj_writer.add_geometries(leaf.feature_list, offset=distance)
        return distance if user_args.offset[0] == 'centroid' else transform_offset
```

But I'm not sure this fixes reprojection/offset issues

Reply:
I will try it out today. In my case OBJ meshes are coming in different crs. Most of them are in US State Plane coordinate systems and UTM. These  OBJ files look like this:

# 500926 vertices, 1000000 faces
mtllib Phantom_HighRes_v2_simplified_3d_mesh.mtl
v -923.414 -502.832 -151.502
v -953.754 -418.45 -151.468
v -925.138 -503.514 -151.66
v 968.921 -842.595 -152.199
v -344.823 -977.591 -152.428
v -970.804 -431.774 -113.181
v -253.187 -1085.63 -151.592
...............................................

I also get a separate offset.xyz file with this content:
664858.000 1342903.000 548.000

If I add offset to vertices coordinates, I will get real world coordinates in input crs.

In my project I am overlapping vector geometries on top of 3D tiles. Vector geometries are coming in geojson file. That's why I need to have reprojection code work well and without my changes I was getting discrepancy ~2 meters or so.